apps/api/src/domain/models/song/song.entity.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    IsNonNegativeFiniteNumber,
+    IsOptional,
+    IsStringWithNonzeroLength,
+} from '@coscrad/validation';
+import { InternalError } from '../../../lib/errors/InternalError';
+import cloneToPlainObject from '../../../lib/utilities/cloneToPlainObject';
+import isNumberWithinRange from '../../../lib/validation/geometry/isNumberWithinRange';
+import { DTO } from '../../../types/DTO';
+import InconsistentTimeRangeError from '../../domainModelValidators/errors/context/invalidContextStateErrors/timeRangeContext/InconsistentTimeRangeError';
+import { Valid } from '../../domainModelValidators/Valid';
+import { resourceTypes } from '../../types/resourceTypes';
+import { TimeRangeContext } from '../context/time-range-context/time-range-context.entity';
+import { Resource } from '../resource.entity';
+
+type ContributorAndRole = {
```

Comment:
Make this a class. It's important to do so for nested 'simple invariant' validation.

Reply:
would this be like this or do we need a constructor?

class ContributorAndRole = {
    contributorId: string;
    role: string;
}